### ISSUE TITLE
Android 3.5.1 and 3.5.2 are broken on the custom url +  comm stats build

### ISSUE THREAD

**Issue by biblicabeebli** [onnela-lab/beiwe-android#70]
There is a bad bug in 3.5.1.
- The app doesn't load, but only if you downloaded the apk it from a link on a Beiwe webpage.
- I just pushed a version 3.5.2a that should now download when you click that link. If the link still downloads 3.5.1 try loading the page in an incognito tab and clicking the link. (the browser caches a redirect, I think.)
- The 3.5.1 Play Store version of the app does not have this bug - I don't know why it doesn't have this bug. It is possible 
- Play store 3.5.1 may crash if you have bluetooth enabled on a study, but it doesn't crash at load.
- 3.5.2a has the bluetooth data stream completely disabled and will probably not be released to the play store.

```
FATAL EXCEPTION: main
Process: org.beiwe.app, PID: 17896
java.lang.RuntimeException: Unable to resume activity {org.beiwe.app/org.beiwe.app.ui.registration.RegisterActivity}: java.lang.SecurityException: Settings key: <bluetooth_address> is only readable to apps with targetSdkVersion lower than or equal to: 31
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4773)
at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4806)
at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2306)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loopOnce(Looper.java:201)
at android.os.Looper.loop(Looper.java:288)
at android.app.ActivityThread.main(ActivityThread.java:7918)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: java.lang.SecurityException: Settings key: <bluetooth_address> is only readable to apps with targetSdkVersion lower than or equal to: 31
at android.provider.Settings$NameValueCache.getStringForUser(Settings.java:3081)
at android.provider.Settings$Secure.getStringForUser(Settings.java:6191)
at android.provider.Settings$Secure.getString(Settings.java:6147)
at org.beiwe.app.DeviceInfo.initialize(DeviceInfo.kt:44)
at org.beiwe.app.ui.registration.RegisterActivity.onResume(RegisterActivity.java:233)
at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1564)
at android.app.Activity.performResume(Activity.java:8474)
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4763)
at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4806) 
at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57) 
at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:179) 
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2306) 
at android.os.Handler.dispatchMessage(Handler.java:106) 
at android.os.Looper.loopOnce(Looper.java:201) 
at android.os.Looper.loop(Looper.java:288) 
at android.app.ActivityThread.main(ActivityThread.java:7918) 
at java.lang.reflect.Method.invoke(Native Method) 
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
W  java.lang.RuntimeException: Unable to resume activity {org.beiwe.app/org.beiwe.app.ui.registration.RegisterActivity}: java.lang.SecurityException: Settings key: <bluetooth_address> is only readable to apps with targetSdkVersion lower than or equal to: 31
W  	at org.beiwe.app.DeviceInfo.initialize(DeviceInfo.kt:44)
W  	at org.beiwe.app.ui.registration.RegisterActivity.onResume(RegisterActivity.java:233)
{32fc4a6 u0 Splash Screen org.beiwe.app EXITING}: Setting back callback null
Input channel object '32fc4a6 Splash Screen org.beiwe.app (client)' was disposed without first being removed with the input manager!
thrown during dispatchAppVisibility Window{28e4fd5 u0 org.beiwe.app/org.beiwe.app.ui.LoadingActivity EXITING}
android.os.DeadObjectException
at android.os.BinderProxy.transactNative(Native Method)
at android.os.BinderProxy.transact(BinderProxy.java:584)
at android.view.IWindow$Stub$Proxy.dispatchAppVisibility(IWindow.java:536)
at com.android.server.wm.WindowState.sendAppVisibilityToClients(WindowState.java:3478)
at com.android.server.wm.WindowContainer.sendAppVisibilityToClients(WindowContainer.java:1234)
at com.android.server.wm.WindowToken.setClientVisible(WindowToken.java:392)
at com.android.server.wm.ActivityRecord.setClientVisible(ActivityRecord.java:6825)
at com.android.server.wm.ActivityRecord.onAnimationFinished(ActivityRecord.java:7647)
at com.android.server.wm.WindowContainer.doAnimationFinished(WindowContainer.java:3394)
at com.android.server.wm.WindowContainer.onAnimationFinished(WindowContainer.java:3406)
at com.android.server.wm.WindowContainer$$ExternalSyntheticLambda4.onAnimationFinished(Unknown Source:2)
at com.android.server.wm.SurfaceAnimator.lambda$getFinishedCallback$0(SurfaceAnimator.java:133)
at com.android.server.wm.SurfaceAnimator.$r8$lambda$lRxTVOJy8fX752UbrFno9INW9hE(Unknown Source:0)
at com.android.server.wm.SurfaceAnimator$$ExternalSyntheticLambda1.run(Unknown Source:8)
at com.android.server.wm.SurfaceAnimator.lambda$getFinishedCallback$1(SurfaceAnimator.java:143)
at com.android.server.wm.SurfaceAnimator.$r8$lambda$4PiCdaEsT4mA6LQVhqpeM5EoU9c(Unknown Source:0)
at com.android.server.wm.SurfaceAnimator$$ExternalSyntheticLambda0.onAnimationFinished(Unknown Source:4)
at com.android.server.wm.RemoteAnimationController.onAnimationFinished(RemoteAnimationController.java:316)
at com.android.server.wm.RemoteAnimationController.-$$Nest$monAnimationFinished(Unknown Source:0)
at com.android.server.wm.RemoteAnimationController$FinishedCallback.onAnimationFinished(RemoteAnimationController.java:432)
at android.view.IRemoteAnimationFinishedCallback$Stub.onTransact(IRemoteAnimationFinishedCallback.java:89)
at android.os.Binder.execTransactInternal(Binder.java:1285)
at android.os.Binder.execTransact(Binder.java:1244)
1969-12-31 19:00:00.000     0-0     <no-tag>                                                     I  ---------------------------- PROCESS STARTED (17896) for package org.beiwe.app ----------------------------
executing service: ServiceRecord{300c72e u0 org.beiwe.app/.MainService}
.beiwe.app                        I  Thread[4,tid=17902,WaitingInMainSignalCatcherLoop,Thread*=0xb40000775c9e62c0,peer=0x13240000,"Signal Catcher"]: reacting to signal 3
.beiwe.app                        I  
.beiwe.app                        I  Wrote stack traces to tombstoned
in org.beiwe.app
PID: 17896
Reason: executing service org.beiwe.app/.MainService
ErrorId: a7cccd7b-c587-4c39-84f6-f0900089b108
Frozen: false
Load: 1.01 / 1.87 / 1.66
----- Output from /proc/pressure/memory -----
some avg10=0.00 avg60=0.00 avg300=0.00 total=12663859
full avg10=0.00 avg60=0.00 avg300=0.00 total=1404386
----- End output from /proc/pressure/memory -----
```

**Comment by biblicabeebli** [comment]
opens to a white screen and then ANRs

**Comment by biblicabeebli** [comment]
Ok there's something that changed with permissions, bluetooth currently crashes the app.... but only if its install via apk file / is the comm stats custom url build. the bug magically does not affect the play store version of the app.  Will have to test a bluetooth enabled study on the play store version to confirm.

**Comment by biblicabeebli** [comment]
description updated with current understanding of the bug and information about the current releases.

**Comment by biblicabeebli** [comment]
The bug directly crashing the app is in accessing the bluetooth MAC - this has been forced to 00:00... on most devices for ages - as part of device info at app start.

**Comment by biblicabeebli** [comment]
Fix implemented on 33d7095e8f8d07236caf9156e4fd0e499947ff4e

**Comment by biblicabeebli** [comment]
Deployed as 3.5.3 on to download links from beiwe servers.
